appservice/src/tree/AppSettingsTreeItem.ts

```diff
@@ -109,7 +109,15 @@ export class AppSettingsTreeItem extends AzureParentTreeItem<ISiteTreeRoot> {
 
         showCreatingTreeItem(newKey);
         settings.properties[newKey] = newValue;
-        await this.root.client.updateApplicationSettings(settings);
+
+        try {
+            await this.root.client.updateApplicationSettings(settings);
+        } catch (err) {
+            // if this is rejected, we need to clear the property otherwise it will cause every subsequent add to fail
+            delete settings.properties[newKey];
```

Comment:
I'm guessing `editSettingItem` (which is used for "edit" and "rename" of an already existing setting) will have a similar problem

Reply:
Good point